porting/build_and_boot/standalone_kernel_build.rst

```diff
@@ -0,0 +1,464 @@
+
+``deviceinfo_kernel_disable_modules="true"``
+
+
+* This option installs all overlay files (in overlay/ directory) into /system/opt/halium-overlay, therefore overlaying the files onto the files present in rootfs instead of completely replacing them. This is necessary for 20.04 and onwards, but not applicable for 16.04.
+
+``deviceinfo_use_overlaystore="true"``
+
+
+* This option is mostly used on some Qualcomm devices. Adding this to your deviceinfo will combine all your dtbo's with your dtbs. Only use if you are sure you need this.
+
+``deviceinfo_kernel_apply_overlay="true"``
+
+
+* Add this configuration to your deviceinfo when you need to specify a fixed size for your rootfs being flashed to system image. Default is 3000M when option is not set. Only use if you need it.
+
+``deviceinfo_system_partition_size="4000M"``
+
+
+* This config is only relevant for newer devices launched with UFS storage
+
+``deviceinfo_rootfs_image_sector_size="4096"``
+
+
+* Next up are flash offsets. You can ignore these if you're building only boot.img for a GKI device, but are required if you want to build vendor_boot.img. These offsets will automatically apply for vendor_boot if your header version is greater than or equal to 3.
+  A guide to extract them is as follows:
+
+Extracting values from stock boot.img/vendor_boot.img
+"""""""""""""""""""""""""""""""""""""""""""""""""""""
+
+For any port to boot, some values must be pulled from the stock boot.img. First and foremost, try to get your stock firmware. Some manufacturers like Samsung like to encrypt their firmwares, but still some tools are available to get them. Others like Xiaomi provide the firmware, but you need to use Google to search it. And then there's the good guy Google, who provide full firmware binaries easily accessible. Long story short, you need to at least find your stock firmware, and then get the boot.img. Once you have the boot.img acquired, follow the following steps:
+
+
+    * Make a directory in which you'll store all your unpacked data. It can be anywhere, and can have any name. For now, we'll call it `temp`.
+
+    * Run "mkdir ~/temp" to create the directory and "cd ~/temp" to go to it.
+
+    * Now to unpack the boot image there are several utilities to do so. One of them is the python script provided by Android. For now we'll use this. To download it into your current working directory run "wget https://raw.githubusercontent.com/LineageOS/android_system_tools_mkbootimg/lineage-19.1/unpack_bootimg.py" This will download the unpack_bootimg.py script, which you'll now use to pull your kernel offsets and other values alike
+
+    * Bring your boot.img into the temp directory.
+
+    * Now run the script. The usual syntax is "python3 unpack_bootimg.py --boot_img <boot.img or vendor_boot.img> --out out" This will unpack the boot image, store the output files in the out directory, and it will also print the offsets on screen.
```

Comment:
Run it with --format mkbootimg and it will provide all the values we need to reconstruct the image as arguments

Reply:
Done